File: 프로그래머스/0/120847. 최댓값 만들기 （1）/최댓값 만들기 （1）.java
class Solution {
    public int solution(int[] numbers) {
        int answer = 0;
        int check = 0;              // 원소 중복 체크 인덱스
        int max1 = -1, max2 = -1;   // 원소1, 원소2

        // 원소1 구하기
        for(int i=0; i<numbers.length; i++) {
            if(max1 < numbers[i]) {
                max1 = numbers[i];
                check = i;
            }
        }
        
        // 원소2 구하기(원소1과 중복되지 않는지 체크)
        for(int i=0; i<numbers.length; i++) {
            if((max2 < numbers[i]) && (check != i)) {
                max2 = numbers[i];
            }
        }
        
        // 최댓값 구하기
        answer = max1 * max2;
        
        return answer;
    }
}

File: 프로그래머스/0/120847. 최댓값 만들기 （1）/최댓값 만들기 （1）.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

// numbers_len은 배열 numbers의 길이입니다.
int solution(int numbers[], size_t numbers_len) {
    int answer = 0;
    int m1 = -1, m2 = -1;
    
    for(int i=0; i<numbers_len; i++) {
        if(m1 < numbers[i]) m1 = numbers[i];    
    }
    for(int i=0; i<numbers_len; i++) {
        if((m2 < numbers[i]) && (m1 != numbers[i])) m2 = numbers[i];
    }    
    if(m2 == -1) m2 = m1;
    
    answer = m1 * m2;
    
    return answer;
}